Counterexamples/VitaliSetNotMeasurable.lean

```diff
@@ -0,0 +1,318 @@
+/-
+Copyright (c) 2024-present Ching-Tsun Chou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ching-Tsun Chou <chingtsun.chou@gmail.com>
+-/
+
+import Mathlib.MeasureTheory.Measure.Lebesgue.Basic
+import Mathlib.MeasureTheory.Measure.NullMeasurable
+
+/-!
+# Vitali set and its non-measurability
+
+This file defines the Vitali set and proves that it is not measurable.
+The proof is essentially the one in Wikipedia:
+
+## References
+
+* <https://en.wikipedia.org/wiki/Vitali_set>
+-/
+
+open Set Filter Real MeasureTheory
+
+noncomputable section
+
+/-- We first enumerate all measure theoretic results that we will need.  We do so
+because we want to make clear exactly which properties of measurability and the volume
+measure are used in the proof of the non-measurability of the Vitali set. -/
+
+example {a b : ℝ} : volume (Icc a b) = ENNReal.ofReal (b - a) :=
+  volume_Icc
+
+lemma volume_mono {s t : Set ℝ} (h : s ⊆ t) : volume s ≤ volume t := by
+  exact OuterMeasureClass.measure_mono volume h
+
+lemma measurable_nullmeasurable {s : Set ℝ} (h : MeasurableSet s) : NullMeasurableSet s volume :=
+  h.nullMeasurableSet
+
+lemma measurable_null_nullmeasurable {s t : Set ℝ}
+    (hm : NullMeasurableSet s) (hn : volume t = 0) : NullMeasurableSet (s ∪ t) :=
+  NullMeasurableSet.union_null hm hn
+
+lemma nullmeasurable_measurable_null {s : Set ℝ} (h : NullMeasurableSet s volume) :
+    ∃ t ⊆ s, MeasurableSet t ∧ volume t = volume s ∧ volume (s \ t) = 0 := by
+  have ⟨t, t_sub_s, t_m, t_eq_s⟩ := h.exists_measurable_subset_ae_eq
+  use t, t_sub_s, t_m
+  constructor
+  · exact measure_congr t_eq_s
+  · exact ae_le_set.mp t_eq_s.symm.le
+
+lemma shift_volume (s : Set ℝ) (c : ℝ) : volume ((fun x ↦ x + c) '' s) = volume s := by
+  simp only [image_add_right, measure_preimage_add_right]
+
+lemma shift_nullmeasurable {s : Set ℝ} (h : NullMeasurableSet s volume) (c : ℝ) :
+    NullMeasurableSet ((fun x ↦ x + c) '' s) volume := by
+  rcases nullmeasurable_measurable_null h with ⟨t, ts, tm, vs, vt⟩
+  rw [← union_diff_cancel ts, image_union]
+  refine measurable_null_nullmeasurable ?_ ?_
+  · have : MeasurableSet ((fun x ↦ x + c) '' t) := by
```

Comment:
A general code style suggestion: you can simplify this by combining the apply and exact:
- first, inline last line into the `apply` (because both are reasonably short)
`by exact (MeasurableEmbedding.measurableSet_image (measurableEmbedding_addRight c)_).mpr tm`.
- then, you can in fact use dot notation: 
`by exact ((measurableEmbedding_addRight c).measurableSet_image).mpr tm`
- finally, you can remove the `by exact`, i.e. write `((measurableEmbedding_addRight c).measurableSet_image).mpr tm`.

Then, I would inline this `have` (i.e., replace `this` by the proof term above). `measurable_nullmeasurable` makes clear what the second argument must prove: there's no need to state this as an extra have.